c++/src/kj/common.h

```diff
@@ -1932,6 +1932,17 @@ class ArrayPtr: public DisallowConstCopyIfNotConst<T> {
   // Syntax sugar for invoking U::from.
   // Used to chain conversion calls rather than wrap with function.
 
+  inline void fill(T t) { 
+    // Fill the area by copying t over every element.
+
+    if constexpr (sizeof(T) == 1) {
```

Comment:
This is not necessarily safe, since someone could have a non-trivial class of one-byte size.

IMO we should change this method to strictly zero instead of fill. And then optimize as follows:
- For types that are trivially constructable/destructable, use memset.
- For other types, use a loop assigning to zero.

Reply:
The other approach is to use `std::fill` under the hood. It should include this optimization?